src/app/components/SignupPage/index.js

```diff
@@ -0,0 +1,294 @@
+      value = Boolean(value[0] === 'true');
+    }
+    this.setState({
+      [name]: value,
+    });
+  }
+
+  handleCheck() {
+    const { isAgreed } = this.state;
+    this.setState({
+      isAgreed: !isAgreed,
+    });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const {
+      fullname,
+      username,
+      email,
+      userPassword,
+      confirmUserPassword,
+      address,
+      school,
+      isAgreed,
+    } = this.state;
+    chkFullname = fullname[0] !== '' && fullname[0] !== undefined;
+    chkUsername = username[0] !== '' && username[0] !== undefined;
+    chkEmail = testEmail.test(email[0]);
+    chkPassword = testPassword.test(userPassword[0]);
+    isMatchPassword = userPassword[0] === confirmUserPassword[0];
+    chkAddress = address[0] !== '' && address[0] !== undefined;
+    chkSchool = school[0] !== '' && school[0] !== undefined;
+    chkAgreed = isAgreed;
+    isSuccess =
+      chkFullname &&
+      chkUsername &&
+      chkEmail &&
+      chkPassword &&
+      isMatchPassword &&
+      chkAddress &&
+      chkSchool &&
+      chkAgreed;
+    isPopup = true;
+    this.forceUpdate();
+  }
+
+  handlePopup() {
+    if (isPopup) {
+      isPopup = false;
+      if (isSuccess) {
+        const {
+          fullname,
+          username,
+          email,
+          userPassword,
+          confirmUserPassword,
+          address,
+          school,
+          participationForm,
+          isAgreed,
+          isStudent,
+        } = this.state;
+        return (
+          <Success
+            fullname={fullname}
+            username={username}
+            email={email}
+            userPassword={userPassword}
+            confirmUserPassword={confirmUserPassword}
+            address={address}
+            school={school}
+            participationForm={participationForm}
+            isAgreed={isAgreed}
+            isStudent={isStudent}
+          />
+        );
+      }
+      return <Failed />;
+    }
+    return '';
+  }
+
+  render() {
+    const { isAgreed } = this.state;
+    return (
+      <div>
+        <Header username="" />
+        <div className={styles.signupForm}>
+          <div className={isPopup === true ? styles.popup : ''}>{this.handlePopup()}</div>
+          <form onSubmit={this.handleSubmit}>
+            <div className={styles.titleBar}>
+              <h5>Đăng ký tham gia FYT Code Cup</h5>
+              <div className={styles.reminder}>
+                <p>Để tham gia FYT Code Cup, hãy đăng ký trước 19:30 ngày 09/10/2020 bạn nhé!</p>
+              </div>
+            </div>
+            <div className={styles.content}>
+              <div className={styles.fullname}>
+                <small>Họ và tên</small> <br />
+                <input type="text" name="fullname" onChange={this.handleChange} />
+                <div className={styles.errorMessage}>
+                  <ShowError chk={chkFullname} s1="" s2="Xin vui lòng điền thông tin" />
+                </div>
```

Comment:
Abstract these input fields into components.

You are having a lot of duplications here.

Reply:
e.g.

```
function InputText({ label, name, onChange, value }) {
...
}
```